### DISCUSSION TITLE
Custom Field of type event to show up in Approved bookings list

### DISCUSSION THREAD

**Discussion by meilensteindigital** [liedekef/events-made-easy#164] [General]
i have set up a custom field as a type of “eventfield” and it is displayed correctly in the events custom field section. i want to use this field as a custom ID for events. 

how can i make this field selectable in the approved bookings table, so i can export this information as well.

for now only booking or person related fields are selectable.

thanks in advance
Thomas

**Comment by meilensteindigital** [comment]
any help is appreciated!

**Comment by liedekef** [comment]
Currently this is not possible via the printable booking overview. However, you can use the filters to change the CSV output (when clicking on the CSV link below the event name in the backend), see the filters doc on that: https://www.e-dynamics.be/wordpress/category/documentation/12-hooks-and-filters/ : eme_csv_column_filter
Any other method requires code changes ...

**Comment by meilensteindigital** [comment]
Hi Franky,Thank you for your answer. Could you please give me an example, how to call a custom field from $events and add this to the csv output

**Reply by meilensteindigital** [reply]
Or could someone else point me in the right direction, please?

**Reply by liedekef** [reply]
I updated the doc a bit. From the hooks doc:

- get the answer for a member by calling: <code>$answers = eme_get_member_answers($member_id);</code>
- get the answer for a person by calling: <code>$answers = eme_get_person_answers($person_id);</code>
- get the answer for a booking by calling: <code>$answers = eme_get_booking_answers($booking_id);</code>
- get the answer for a event by calling: <code>$answers = eme_get_event_answers($event_id);</code>
- get the answer for a location by calling: <code>$answers = eme_get_location_answers($location_id);</code>
- get the answer for a membership by calling: <code>$answers = eme_get_membership_answers($membership_id);</code>

Now, adding something from an event to the CSV or booking report is best done via the filter eme_csv_header_filter. On the hooks page https://www.e-dynamics.be/wordpress/hooks-and-filters/ you can see some examples. 

Looping through the answers is done via foreach. Below an example to just print out on stdout the answer for field with field id MY_FIELD_ID (to be replaced with the field id of choice):
```
        foreach ($answers as $answer) {
            if ($answer['field_id'] == MY_FIELD_ID) {
               print $answer['answer'];
            }
        }
```
If you want to filter on field name and also convert the answer to readable output (for ex. for dropdown), you can use:
```
        foreach ($answers as $answer) {
            $formfield = eme_get_formfield($answer['field_id'];
            if ($formfield['field_name'] == MY_FIELD_NAME) {
               print eme_answer2readable($answer['answer'],$formfield);
            }
        }
```
The function eme_answer2readable has also options to change the separator etc ... but that I'll leave to your testing.

**Comment by meilensteindigital** [comment]
Hi Franky, thank you for bringing this feature to the current version of EME. But i have a small issue there. In my setup for each event there is a custom textfield which holds just text, but when i activate this field in the approved bookings overview, text inputs from other events are shown as well.

<img width="1387" alt="image" src="https://github.com/liedekef/events-made-easy/assets/93185505/96ce4e6a-d07a-4a19-a667-975af5f6855f">
<img width="536" alt="image" src="https://github.com/liedekef/events-made-easy/assets/93185505/481a46cf-c9ae-40f0-94f7-fdfdd0df7883">

could you please have a look at this.

Thank you 
Thomas

**Reply by meilensteindigital** [reply]
does anyone else have an idea  why this is happening?

**Comment by meilensteindigital** [comment]
by the way .. it works with the event listing

<img width="1458" alt="image" src="https://github.com/liedekef/events-made-easy/assets/93185505/0fb7a4c2-7ced-4b18-bcec-a781bde99cda">

**Comment by liedekef** [comment]
This is in fact a bug (and I forgot to reply, holidays you know ...).
I already found why, I'll come up with a fix today.

**Reply by meilensteindigital** [reply]
that's awesome, Franky

**Reply by liedekef** [reply]
Please try the eme-rsvp.php version from this change:
https://github.com/liedekef/events-made-easy/commit/fc360c5b9bae999874078b18bd1b3376d1abc9c9

**Reply by liedekef** [reply]
sorry, fix not complete yet :-)

**Reply by liedekef** [reply]
This extra typo fix is needed: https://github.com/liedekef/events-made-easy/commit/c89730b8f466f6afad9908c0c43110127f186b25

**Reply by meilensteindigital** [reply]
it works, thank you so much !
